walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,205 @@
 package com.google.sps;
 
 import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.ArrayList;
 
 public final class FindMeetingQuery {
+
+  // Constants for 0 and the number of minutes in a day. 
+  // 0 means 0 minute since start of a day. 
+  // 1440 is the total number of minutes in a day, which marks the end of a day. 
+  private static final int START_OF_DAY = 0; 
+  private static final int END_OF_DAY = 1440; 
+
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+
+    Collection<String> mandatoryAttendees = request.getAttendees();
+    Collection<String> optionalAttendees = request.getOptionalAttendees(); 
+    long meetingDuration = request.getDuration(); 
+
+    List<TimeRange> mandatoryAttendeesBusyTimeslots = getAllBusyTimeslots(events, mandatoryAttendees); 
+    List<TimeRange> optionalAttendeesBusyTimeslots = getAllBusyTimeslots(events, optionalAttendees);
+
+    sortBusyTimeslotsByStart(mandatoryAttendeesBusyTimeslots); 
+    sortBusyTimeslotsByStart(optionalAttendeesBusyTimeslots);
+
+    List<TimeRange> mandatoryAttendeesEmptySlots = getEmptySlots(mandatoryAttendeesBusyTimeslots, meetingDuration); 
+    List<TimeRange> optionalAttendeesEmptySlots = getEmptySlots(optionalAttendeesBusyTimeslots, meetingDuration);
+
+    if (mandatoryAttendees.size() == 0) {
```

Comment:
When you're checking emptiness we can call isEmpty()

if (mandatoryAttendees.isEmpty()) {
  ...

Reply:
Updated.